server/map_rpc_server.go

```diff
@@ -263,6 +267,23 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 				HashedValue: l.LeafHash,
 			})
 		}
+
+		if t.opts.UseSingleTransaction && t.opts.UseLargePreload {
+			readRev, err := tx.ReadRevision(ctx)
+			if err != nil {
+				return err
+			}
+			nids := make([]storage.NodeID, 0, len(hkv)*256)
```

Comment:
Can any of these IDs be duplicates?

Reply:
They can. I figured that the storage layer will dedupe, but you're right - it's best to be explicit and do it here [too].

Reply:
Yeah storage will definitely dedupe but I think the map could get fairly large here so worth doing.

Reply:
(done)